File: src/dataset/go/bottle_song_Exercism_go_rootulp/gemini/GeminiFlash_bottle_song_v1.go
package bottlesong

import (
	"fmt"
	"strings"
)

var numberToCardinal = map[int]string{
	10: "ten",
	9:  "nine",
	8:  "eight",
	7:  "seven",
	6:  "six",
	5:  "five",
	4:  "four",
	3:  "three",
	2:  "two",
	1:  "one",
	0:  "no",
}

func Recite(startBottles, takeDown int) []string {
	song := make([]string, 0, takeDown*5) // Pre-allocate memory for the song
	for i := startBottles; i > startBottles-takeDown; i-- {
		verseLines := verse(i)
		song = append(song, verseLines...)
		if i > startBottles-takeDown+1 { // Avoid appending "" to the last verse
			song = append(song, "")
		}
	}
	return song
}

func verse(bottles int) []string {
	startCardinal := cardinal(bottles)
	endCardinal := cardinal(bottles - 1)
	pluralStart := maybePluralize(bottles, "bottle", "bottles")
	pluralEnd := maybePluralize(bottles-1, "bottle", "bottles")
	startCap := capitalize(startCardinal)

	return []string{
		fmt.Sprintf("%s green %s hanging on the wall,", startCap, pluralStart),
		fmt.Sprintf("%s green %s hanging on the wall,", startCap, pluralStart),
		"And if one green bottle should accidentally fall,",
		fmt.Sprintf("There'll be %s green %s hanging on the wall.", endCardinal, pluralEnd),
	}
}

func cardinal(number int) string {
	return numberToCardinal[number]
}

func capitalize(str string) string {
	if len(str) == 0 {
		return ""
	}
	// Convert string to rune slice to handle UTF-8 characters correctly
	runes := []rune(str)
	runes[0] = []rune(strings.ToUpper(string(runes[0])))[0] // Capitalize the first rune
	return string(runes)
}

func maybePluralize(number int, singularWord string, pluralWord string) string {
	if number == 1 {
		return singularWord
	}
	return pluralWord
}

File: src/dataset/Java/bottle_song_Exercism_java_rabestro/gemini/GeminiFlash_BottleSong_v1.java
import java.util.function.IntFunction;
import java.util.stream.IntStream;

import static java.util.stream.Collectors.joining;

class BottleSong {

    private static final String[] NUMBER_WORDS = {"no", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"};

    private String firstHalf(int bottles) {
        String bottleText = (bottles == 1) ? "bottle" : "bottles";
        String numberWord = (bottles >= 0 && bottles <= 10) ? NUMBER_WORDS[bottles] : String.valueOf(bottles); // Handle cases outside 0-10 if needed
        return numberWord.substring(0, 1).toUpperCase() + numberWord.substring(1) + " green " + bottleText + " hanging on the wall,\n";
    }

    private String lastHalf(int bottles) {
        String bottleText = (bottles == 1) ? "bottle" : "bottles";
        String numberWord = (bottles >= 0 && bottles <= 9) ? NUMBER_WORDS[bottles] : String.valueOf(bottles); // Handle cases outside 0-9 if needed

        return "And if one green bottle should accidentally fall,\n" +
                "There'll be " + numberWord + " green " + bottleText + " hanging on the wall.\n";
    }

    private String verse(int bottles) {
        return firstHalf(bottles) + firstHalf(bottles) + lastHalf(bottles - 1);
    }

    String recite(int startBottles, int takeDown) {
        return IntStream.rangeClosed(startBottles - takeDown + 1, startBottles)
                .boxed()
                .sorted((a, b) -> b - a)
                .map(this::verse)
                .collect(joining(System.lineSeparator()));
    }
}